adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenRequestTest.java

```diff
@@ -554,6 +559,306 @@ public void testVerifyBrokerRedirectUriInvalidSignature()
+        final Calendar expiredTime = new GregorianCalendar();
+        expiredTime.add(Calendar.MINUTE, -MINUS_MINUITE);
+        final Calendar extendedExpiresTime = new GregorianCalendar();
+        extendedExpiresTime.add(Calendar.MINUTE, -1);
+        final ITokenCacheStore cacheStore = getTokenCache(expiredTime.getTime(), false, false, extendedExpiresTime.getTime());
+
+        final FileMockContext mockContext = createMockContext();
+        AuthenticationSettings.INSTANCE.setUseBroker(false);
+        final AuthenticationContext authContext = new AuthenticationContext(mockContext,
+                VALID_AUTHORITY, false, cacheStore);
+        //turn on the outage mode
+        authContext.setExtendedLifetimeEnabled(true);
+        //mock http response
+        final HttpURLConnection mockedConnection = Mockito.mock(HttpURLConnection.class);
+        HttpUrlConnectionFactory.mockedConnection = mockedConnection;
+        Util.prepareMockedUrlConnection(mockedConnection);
+        Mockito.when(mockedConnection.getOutputStream()).thenReturn(Mockito.mock(OutputStream.class));
+        Mockito.when(mockedConnection.getInputStream()).thenReturn(Util.createInputStream(Util.getSuccessTokenResponse(false, false)));
+        Mockito.when(mockedConnection.getResponseCode()).thenReturn(HttpURLConnection.HTTP_GATEWAY_TIMEOUT);
+
+        try {
+            authContext.acquireTokenSilentSync("resource", "clientid", TEST_USERID);
+            fail("Not expected exception.");
+        } catch (final Exception exception) {
+            //verify the retry
+            verify(mockedConnection, times(2)).getInputStream();
+            assertNotNull(exception);
+            assertTrue(exception instanceof AuthenticationException);
+            assertTrue(((AuthenticationException) exception).getCode() == ADALError.AUTH_FAILED_NO_TOKEN);
+        } finally {
+            cacheStore.removeAll();
+        }
+    }
+
+    /**
+     * Test for returning new AT from the server when the
+     * outage mode is on and the stale AT is valid
+     */
+    @SmallTest
+    public void testResiliencyTokenReturn_outageModeOn_validAT() throws PackageManager.NameNotFoundException,
+            NoSuchAlgorithmException, OperationCanceledException, IOException, AuthenticatorException,
+            InterruptedException {
+        // make sure AT's expires_in is expired and ext_expires_in is not expired
+        final Calendar expiredTime = new GregorianCalendar();
+        expiredTime.add(Calendar.MINUTE, -MINUS_MINUITE);
+        final Calendar extendedExpiresTime = new GregorianCalendar();
+        extendedExpiresTime.add(Calendar.MINUTE, EXTEND_MINUS_MINUTE);
+        final ITokenCacheStore cacheStore = getTokenCache(expiredTime.getTime(), false, false, extendedExpiresTime.getTime());
+
+
+        final FileMockContext mockContext = createMockContext();
+        AuthenticationSettings.INSTANCE.setUseBroker(false);
```

Comment:
Yes, we actually need. Because when other tests set the setUserBroker(true), the setting is changed. If we did not set it back, the test will be broken.